src/vs/editor/common/model/textModel.ts

```diff
@@ -620,24 +620,25 @@ export class TextModel extends Disposable implements model.ITextModel {
 		let guessedIndentation = guessIndentation(this._buffer, defaultTabSize, defaultInsertSpaces);
 		this.updateOptions({
 			insertSpaces: guessedIndentation.insertSpaces,
-			tabSize: guessedIndentation.tabSize
+			tabSize: guessedIndentation.tabSize,
+			indentSize: guessedIndentation.tabSize, // TODO@Alex: guess indentSize independent of tabSize
 		});
 	}
 
-	private static _normalizeIndentationFromWhitespace(str: string, tabSize: number, insertSpaces: boolean): string {
+	private static _normalizeIndentationFromWhitespace(str: string, indentSize: number, insertSpaces: boolean): string {
 		let spacesCnt = 0;
 		for (let i = 0; i < str.length; i++) {
 			if (str.charAt(i) === '\t') {
-				spacesCnt += tabSize;
+				spacesCnt += indentSize;
```

Comment:
This doesn't look right. Shouldn't a tab character always use `tabSize`, not `indentSize`?

Reply:
The callers of that method usually append a `\t` to indicate increase indentation. i.e. callers of this method are command implementations.

I don't think this is related to https://github.com/Microsoft/vscode/issues/10339#issuecomment-467242102 , I believe that one is a user configuration error.